lua/neotest/consumers/run.lua

```diff
@@ -13,6 +13,11 @@ local neotest = {}
 --- A consumer providing a simple interface to run tests.
 neotest.run = {}
 
+---@private
```

Comment:
Similar comment, we can't treat this as private if another consumer is using it.

I think we should do the unpacking (and thus nil check) here as it'll be a more consistent API. 

Please also add a doc message with a description and return values, the Github Actions will generate a commit to update the docs with it.

Reply:
Done. Still seems a bit strange to have a public `.run()` that returns a `tree, adapter_id`?